service/storage_service.cc

```diff
@@ -1817,7 +1822,7 @@ future<> topology_coordinator::run() {
             if (!had_work) {
                 // Nothing to work on. Wait for topology change event.
                 slogger.trace("raft topology: topology coordinator fiber has nothing to do. Sleeping.");
-                co_await _topo_sm.event.when();
+                co_await await_event();
                 slogger.trace("raft topology: topology coordinator fiber got an event");
```

Comment:
I suggest to restructure the loop later in the conventional condition variable style.